### ISSUE TITLE
Async Error: "The async task did not return valid JSON"

### ISSUE THREAD

**Issue by nhooey** [ansible/ansible#14714]
This issue was originally covered in the [mailing list thread](https://groups.google.com/forum/#!topic/ansible-devel/9WLR5wpguH8), but I'm re-posting it here as I have the same issue with Ansible 2.0.0.2.
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /home/deploy/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

``` ini
[defaults]
host_key_checking = False
forks = 20
timeout = 20
```
##### Environment:

CentOS release 6.6 (Final)
##### Summary:

Running an `async` task renders the following error:

```
fatal: [hostname]: FAILED! => {"msg": "The async task did not return valid JSON: No JSON object could be decoded", "failed": true, "changed": false}
```
##### Steps To Reproduce:

Run a task with `async` and `poll` set.

``` yaml
    - name: bo1 status
      command: sudo systemctl status data-mongodb-bo1.mount
      ignore_errors: yes
      async: yes
```
##### Expected Results:

A successful Ansible task, since I can verify that the command ran successfully on the machine due to its output being written to a file.
##### Actual Results:

```
TASK [bo1 status] ***************************************************************
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 aws-beta-coreos-01 mkdir -p $HOME/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257 && echo $HOME/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> PUT /tmp/tmp8ZxCcg TO /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/command
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 aws-beta-coreos-01 chmod a+rx /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/command
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> PUT /tmp/tmpLnru7f TO /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/async_wrapper
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 aws-beta-coreos-01 chmod a+rx /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/async_wrapper
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> PUT /tmp/tmpKf5dpr TO /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/arguments
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 aws-beta-coreos-01 /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/async_wrapper 471364022938 1 /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/command /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/arguments
<aws-beta-coreos-01> ESTABLISH SSH CONNECTION FOR USER: core
<aws-beta-coreos-01> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 aws-beta-coreos-01 rm -rf /home/core/.ansible/tmp/ansible-tmp-1435307794.23-77712038923257/ >/dev/null 2>&1
fatal: [aws-beta-coreos-01]: FAILED! => {"msg": "The async task did not return valid JSON: No JSON object could be decoded", "failed": true, "changed": false}
```

**Comment by nhooey** [comment]
Pinging those of you who were discussing this on the [mailing list thread](https://groups.google.com/forum/#!topic/ansible-devel/9WLR5wpguH8) before.
@youam, @bcoca, @rcarrillocruz, @hgrubst

**Comment by nhooey** [comment]
Do you have any comments on the proposed fix at ansible/ansible#14716, @jimi-c?

**Comment by bcoca** [comment]
dupe of #14568, though this is more explicit on the 'general case', but that will be view able in link
